### ISSUE TITLE
No support for MANAGED location syntax on schemas in Databricks dialect

### ISSUE THREAD

**Issue by TheCleric** [sqlfluff/sqlfluff#6181]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

1. When creating up a SCHEMA with a MANAGED LOCATION option, I get an error: `Found unparsable section: "MANAGED LOCATION ..."`

### Expected Behaviour

SQLFluff supports MANAGED LOCATION according to this spec: https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-schema.html

### Observed Behaviour

An unparsable error is thrown.

### How to reproduce

Use SQLFluff to lint the following SQL

```sql
CREATE SCHEMA IF NOT EXISTS my_schema
MANAGED LOCATION 's3://my-s3-bucket/';
```

### Dialect

Databricks

### Version

sqlfluff, version 3.1.1
Python 3.10.13

### Configuration

```toml
[sqlfluff]
verbose = 1
dialect = databricks
templater = placeholder
max_line_length = 120
exclude_rules = CP02, ST06
warnings = RF03
large_file_skip_byte_limit = 0

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = True
indented_ctes = True
indented_using_on = True
indented_on_contents = True
# indented_then = True
# indented_then_contents = True
allow_implicit_indents = True
template_blocks_indent = True

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
